packages/instrumentation-llamaindex/src/instrumentation.ts

```diff
@@ -0,0 +1,358 @@
+    moduleExports: typeof llamaindex & { openLLMetryPatched?: boolean },
+  ) {
+    this._unwrap(moduleExports.RetrieverQueryEngine.prototype, "query");
+
+    for (const key in moduleExports) {
+      const cls = (moduleExports as any)[key];
+      if (this.isLLM(cls.prototype)) {
+        this._unwrap(cls.prototype, "complete");
+        this._unwrap(cls.prototype, "chat");
+      } else if (this.isEmbedding(cls.prototype)) {
+        this._unwrap(cls.prototype, "getQueryEmbedding");
+      } else if (this.isSynthesizer(cls.prototype)) {
+        this._unwrap(cls.prototype, "synthesize");
+      } else if (this.isRetriever(cls.prototype)) {
+        this._unwrap(cls.prototype, "retrieve");
+      }
+    }
+
+    return moduleExports;
+  }
+
+  private genericWrapper(className: string, methodName: string) {
+    // eslint-disable-next-line @typescript-eslint/no-this-alias
+    const plugin = this;
+    // eslint-disable-next-line @typescript-eslint/ban-types
+    return (original: Function) => {
+      return function method(this: BaseEmbedding, ...args: unknown[]) {
+        const span = plugin.tracer.startSpan(
+          `llamaindex.${className}.${methodName}`,
+        );
+        const execContext = trace.setSpan(context.active(), span);
+        const execPromise = safeExecuteInTheMiddle(
+          () => {
+            return context.with(execContext, () => {
+              return original.apply(this, args);
+            });
+          },
+          (error) => {
+            // if (error) {
+            // }
+          },
+        );
+        const wrappedPromise = execPromise
+          .then((result: any) => {
+            return new Promise((resolve) => {
+              span.setStatus({ code: SpanStatusCode.OK });
+              span.end();
+              resolve(result);
+            });
+          })
+          .catch((error: Error) => {
+            return new Promise((_, reject) => {
+              span.setStatus({
+                code: SpanStatusCode.ERROR,
+                message: error.message,
+              });
+              span.end();
+              reject(error);
+            });
+          });
+        return context.bind(execContext, wrappedPromise as any);
+      };
+    };
+  }
+
+  private completeWrapper({ className }: { className: string }) {
+    // eslint-disable-next-line @typescript-eslint/no-this-alias
+    const plugin = this;
+    // eslint-disable-next-line @typescript-eslint/ban-types
+    return (original: Function) => {
+      return function method(this: llamaindex.LLM, ...args: unknown[]) {
+        const prompt = args[0];
+
+        const span = plugin.tracer.startSpan(
+          `llamaindex.${className}.complete`,
+        );
+
+        span.setAttribute(SpanAttributes.LLM_VENDOR, "llamaindex");
+        span.setAttribute(
+          SpanAttributes.LLM_REQUEST_MODEL,
+          this.metadata.model,
+        );
+        span.setAttribute(SpanAttributes.LLM_REQUEST_TYPE, "complete");
```

Comment:
Should be "completion", but it's best to take it straight from the semantic conventions
https://github.com/traceloop/openllmetry-js/blob/main/packages/ai-semantic-conventions/src/SemanticAttributes.ts#L45